text/0000-pack-toml.md

```diff
@@ -0,0 +1,123 @@
+# Meta
+[meta]: #meta
+- Name: Support for `pack.toml`
+- Start Date: 2021-11-16
+- Author(s): [jkutner](http://github.com/jkutner)
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a `pack.toml` file that is a special case of the `project.toml` descriptor file. The `pack.toml` will include configuration options specific to the Pack CLI, which may not be honored by other platforms.
+
+# Definitions
+[definitions]: #definitions
+
+- *project descriptor* - a extension spec defining the [`project.toml`](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md) file.
+
+
+# Motivation
+[motivation]: #motivation
+
+The `project.toml` file has been a useful and essential tool for buildpack users who need to codify certain arguments of their builds. However, there has been much debate around how to handle configuration options that may not be supported by all platforms.
+
+We desire a descriptor file that is:
+* specific to pack (i.e. not platform-neutral)
+* advertises that the repo containing the file can be built with pack
+* adheres to the best practices and learnings from [`project.toml`](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md)
+
+# What it is
+[what-it-is]: #what-it-is
+
+The Pack CLI will support a new descriptor file, `pack.toml`, that is a superset of the project descriptor.
+* An app MAY have a `pack.toml`
+* An app MAY have a `project.toml`
+* An app MAY have both a `pack.toml` and a `project.toml`, in which case the `project.toml` will be ignored.
```

Comment:
```suggestion
* An app MAY have both a `pack.toml` and a `project.toml` where they will be merged and used.
```

Reply:
@hone we've gone back and forth on this. Merging could actually be very complicated, and if we support merging we should only do it at the top level shared key (i.e. we're not going to try and merge lists of things for example)

Reply:
@jkutner this is precisely why I have a problem with a platform specific descriptor covering platform agnostic configuration.

I would want to have as much as possible in an agnostic way, at least from the perspective of all platforms other than `pack`. Why? Because for 99% of the non `pack` platforms I can imagine, `pack` will be the platform for local execution. See kpack and our integration into project Piper as two examples and Tekton most likely as a third.

Also, as I pointed out in https://github.com/buildpacks/rfcs/pull/189#discussion_r1038109057 the “`project.toml` will be ignored” statement contradicts other parts of this proposal.